File: anacondacode/tensorflow_environment/old_code/计算方法/shiyan1_3.py
# -*- coding: utf-8 -*-
"""
Created on Fri Nov 22 11:12:58 2019

@author: LJZ
"""

###外积
print("tensor product;outer product;外积")
import numpy as np
import matplotlib.pyplot as plt
n=5
a=np.arange(n)
b=np.ones(n)
print("axb,bxa in figure 1")
r1=np.tensordot(a,b,axes=0)
fig=plt.figure()
ax1=plt.subplot(1,2,1)
plt.imshow(r1)
r2=np.tensordot(b,a,axes=0)
ax2=plt.subplot(1,2,2)
plt.imshow(r2)
plt.show()

File: matlab_code/oldcode/shiyan1_3.m
fprintf("tensor product;outer product;Íâ»ý\n");
n=5;
a=0:(n-1);
b=ones(1,n-1);
r1=a'*b;
r2=b'*a;
figure;
subplot(1,2,1);imagesc(r1);
subplot(1,2,2);imagesc(r2);
